plasmapy/particles/tests/test_atomic.py

```diff
@@ -43,345 +43,208 @@
-    ["H-2", "H"],
-    ["D", "H"],
-    ["T", "H"],
-    ["H-3", "H"],
-    ["Hydrogen-3", "H"],
-    ["helium", "He"],
-    [2, "He"],
-    ["alpha", "He"],
-    ["gold", "Au"],
-    ["Gold", "Au"],
-    [79, "Au"],
-    ["79", "Au"],
-    ["P", "P"],
-    [118, "Og"],
-    ["N-14", "N"],
-    ["N", "N"],
-    ["H +1", "H"],
-    ["H 1+", "H"],
-    ["hydrogen 1+", "H"],
-    ["deuterium 1+", "H"],
-    ["Fe 24+", "Fe"],
-    ["Fe +24", "Fe"],
-    ["Fe 2-", "Fe"],
-    ["Fe -2", "Fe"],
-    ["Fe+", "Fe"],
-    ["Fe++", "Fe"],
-    ["Fe-", "Fe"],
-    ["Fe++++++++++++++", "Fe"],
-    ["H-0", InvalidParticleError],
-    [3.14159, TypeError],
-    ["Og-294b", InvalidParticleError],
-    ["H-934361079326356530741942970523610389", InvalidParticleError],
-    ["Fe 2+4", InvalidParticleError],
-    ["Fe+24", InvalidParticleError],
-    ["Fe +59", InvalidParticleError],
-    ["C++++++++++++++++", InvalidParticleError],
-    ["C-++++", InvalidParticleError],
-    ["neutron", InvalidElementError],
-    ["n", InvalidElementError],
-    ["n-1", InvalidElementError],
-    ["h", InvalidParticleError],
-    ["d", InvalidParticleError],
-    ["he", InvalidParticleError],
-    ["au", InvalidParticleError],
-    ["p-", InvalidElementError],
-    [0, InvalidParticleError],
-    [119, InvalidParticleError],
-    ["antiproton", InvalidElementError],
+table_functions_args_kwargs_output = [
+    [atomic_symbol, [1,], {}, "H"],
+    [atomic_symbol, [1], {}, "H"],
+    [atomic_symbol, ["H"], {}, "H"],
+    [atomic_symbol, ["p"], {}, "H"],
+    [atomic_symbol, ["T"], {}, "H"],
+    [atomic_symbol, ["deuterium"], {}, "H"],
+    [atomic_symbol, ["deuteron"], {}, "H"],
+    [atomic_symbol, ["Tritium"], {}, "H"],
+    [atomic_symbol, ["triton"], {}, "H"],
+    [atomic_symbol, ["H-2"], {}, "H"],
+    [atomic_symbol, ["D"], {}, "H"],
+    [atomic_symbol, ["T"], {}, "H"],
+    [atomic_symbol, ["H-3"], {}, "H"],
+    [atomic_symbol, ["Hydrogen-3"], {}, "H"],
+    [atomic_symbol, ["helium"], {}, "He"],
+    [atomic_symbol, [2], {}, "He"],
+    [atomic_symbol, ["alpha"], {}, "He"],
+    [atomic_symbol, ["gold"], {}, "Au"],
+    [atomic_symbol, ["Gold"], {}, "Au"],
+    [atomic_symbol, [79], {}, "Au"],
+    [atomic_symbol, ["79"], {}, "Au"],
+    [atomic_symbol, ["P"], {}, "P"],
+    [atomic_symbol, [118], {}, "Og"],
+    [atomic_symbol, ["N-14"], {}, "N"],
+    [atomic_symbol, ["N"], {}, "N"],
+    [atomic_symbol, ["H +1"], {}, "H"],
+    [atomic_symbol, ["H 1+"], {}, "H"],
+    [atomic_symbol, ["hydrogen 1+"], {}, "H"],
+    [atomic_symbol, ["deuterium 1+"], {}, "H"],
+    [atomic_symbol, ["Fe 24+"], {}, "Fe"],
+    [atomic_symbol, ["Fe +24"], {}, "Fe"],
+    [atomic_symbol, ["Fe 2-"], {}, "Fe"],
+    [atomic_symbol, ["Fe -2"], {}, "Fe"],
+    [atomic_symbol, ["Fe+"], {}, "Fe"],
+    [atomic_symbol, ["Fe++"], {}, "Fe"],
+    [atomic_symbol, ["Fe-"], {}, "Fe"],
+    [atomic_symbol, ["Fe++++++++++++++"], {}, "Fe"],
+    [isotope_symbol, ("He", 4), {}, "He-4"],
```

Comment:
The 4-tuple is pretty exhaustive - for this kind of test, you really don't ever need more info than "function to be tested, positional arguments, keyword arguments and output (exception or value)". I would be in favor of simply standardizing on that - and since you've got the `parametrize` right below, you can look up the current order.

And admittedly, this is not as clean as it could have been - I had some trouble with the macros I was making on the fly and some of the positional args turned into tuples and some into lists. They should probably always be a list.

I don't think the namedtuple helps much. With the keyword arguments in your example, it makes the whole test line a whole lot longer. If we use positional args (`TI(("He", 4), {}, "He-4")` in your example) it stops being informative from the "current line" level. I notice you skipped the kwargs `{}` - but then you have to handle adding them as optional arguments somewhere later on. And in terms of efficiency, wrapping them into a NamedTuple just to unwrap them the next time they're accessed sounds meh at best.

I vote standard test input 4-tuples :)

Reply:
>I don't think the namedtuple helps much. With the keyword arguments in your example, it makes the whole test line a whole lot longer.

I do like the explicitness of a `namedtuple`, but this statement here is what gives me the biggest pause to implementing it package wide.  As of right now, is see major pros and cons with the `namedtuple` route and using `parametrize` in it's native form.  As such, I currently don't see a clear winner.  It would be really nice to find an implementation that is simple and explicit, but when does that ever happen? 

> The 4-tuple is pretty exhaustive - for this kind of test, you really don't ever need more info than "function to be tested, positional arguments, keyword arguments and output (exception or value)". I would be in favor of simply standardizing on that - and since you've got the parametrize right below, you can look up the current order.

I'm not sure this is something that could be standardized.  I think we can standardize a scheme(s) that would work 80-90% of the time, but there will always be a few exceptions.  I'm for this as long as we still leave room for the cases that don't fit the standardization.

The "standard" I typically use is `"args, kwargs, expected"` for a valid call test, `"args, kwargs, _raises"` for an exception test, and `"args, kwargs, _warns"` for a warning test.

Reply:
Okay, that's right. There are multiple levels of generalization. If you're testing one function per `parametrize`, then yes, the 3-tuple you mentioned is optimal. I was thinking about this kind of grouped test for a bunch of functions at the same time, or the ones in test_exceptions.